### ISSUE TITLE
Lint match statement matching #[non_exhaustive] enum

### ISSUE THREAD

**Issue by Pscheidl** [rust-lang/rust-clippy#11890]
### What it does

If a downstream crate/app matches on an enum with  [non_exhaustive](https://doc.rust-lang.org/reference/attributes/type_system.html) attribute set, `rustc` doesn't fail when new field is added to that enum. Therefore, with each dependency update, the existence of new fields must be checked manually. Otherwise they're routed the default way.

This lint would warn if there's a field in an enum marked as `#[non_exhaustive]` that isn't handled by a match statement.

The lint is optional.

### Advantage

- Saves checking codebases manually on crate updates - error prone.
- Makes it easy to spot new errors they potentially want to handle, instead of handling them the "default way".
- 

### Drawbacks

- Some code owners may prefer not to be warned. Must be either implemented as opt-in, or must be able to opt-out.

## Old code
```rust
#[non_exhaustive]
enum Error {
 ValidationError, 
 ResponseError, // Newly added field
}
```

```rust
let error: Error = ... // obtain the error here;
match error { // Warning is trigger for this match statement
 ValidationError => {...},
 _ {
 println!("This is the default path, newly added ResponseError is silently sent this way")
 }
}
```

## Improved match statement

```rust
let error: Error = ... // obtain the error here;
match error { 
 ValidationError => {...},
 ResponseError => {...}, // The new case is handled, clippy no longer warns
 _ {
 println!("This is the default path")
 }
}
```
